### PR TITLE
sundials: add new version

### PR DESCRIPTION
**Pull request by balos1**

### CODE REVIEW

File: var/spack/repos/builtin/packages/sundials/package.py [hunk 1]
```diff
@@ -87,7 +88,7 @@ class Sundials(CMakePackage, CudaPackage, ROCmPackage):
     # Logging
     variant(
         "logging-level",
-        default="0",
+        default="2",
```
Comment (gardner48) [inline var/spack/repos/builtin/packages/sundials/package.py:91]: Is there a way to make the default version dependent so the previous default is retained in older versions?

Comment (gardner48) [reply var/spack/repos/builtin/packages/sundials/package.py:91]: Some other updates below:
* `logging-mpi` variant, `when` can be updated to `when="@6.2.0:6.7.0"`
* `generic-math` variant, add `when="@:6.7.0"`

**Review by gardner48 — CHANGES_REQUESTED** [review]
Some minor variant updates for the new version

**Review by tldahlgren — DISMISSED** [review]
Confirmed the version sha256.

File: var/spack/repos/builtin/packages/sundials/package.py (hunk 1 above)
Comment (gardner48) [reply var/spack/repos/builtin/packages/sundials/package.py:91]: Updated `logging-mpi` and `generic-math`. There does not currently seem to be a way to have the default depend on the version.

**Review by gardner48 — DISMISSED** [review]

**Review by alalazo — CHANGES_REQUESTED** [review]
There are build errors in CI (invalid conversions):
```
 /tmp/root/spack-stage/spack-stage-dealii-9.5.1-glbt6ywp5brskvdbuo2
             zipmupzkvfjtb/spack-src/source/sundials/ida.cc:343:20: error: inva
             lid user-defined conversion from 'dealii::SUNDIALS::IDA<dealii::Ve
             ctor<double> >::reset(double, double, dealii::Vector<double>&, dea
             lii::Vector<double>&)::<lambda(SUNLinearSolver, SUNMatrix, N_Vecto
             r, N_Vector, int)>' to 'int (*)(SUNLinearSolver, SUNMatrix, N_Vect
             or, N_Vector, sunrealtype)' {aka 'int (*)(_generic_SUNLinearSolver
             *, _generic_SUNMatrix*, _generic_N_Vector*, _generic_N_Vector*, do
             uble)'} [-fpermissive]
```

**Comment by balos1** [comment]
> There are build errors in CI (invalid conversions):

@alalazo The errors are coming from deali.ii. I don't think deal.ii's current version is compatible with this new version of SUNDIALS yet. Should the deal.ii package be updated with a constraint?

**Comment by alalazo** [comment]
@balos1 Yes, if dealii has wrong constraints they should be updated in this PR, to avoid breaking builds in CI. When CI is :heavy_check_mark: I'll merge.

File: var/spack/repos/builtin/packages/dealii/package.py [hunk 2]
```diff
@@ -206,7 +206,7 @@ class Dealii(CMakePackage, CudaPackage):
     depends_on("slepc~arpack", when="+slepc+petsc+mpi+int64")
     depends_on("sundials@:3~pthread", when="@9.0:9.2+sundials")
     depends_on("sundials@5:5.8", when="@9.3:9.3.3+sundials")
-    depends_on("sundials@5:", when="@9.3.4:+sundials")
+    depends_on("sundials@5:6.7", when="@9.3.4:+sundials")
```
Comment (gardner48) [inline var/spack/repos/builtin/packages/dealii/package.py:209]: I suspect as similar version constraint will be needed in the MFEM and AMReX packages as well (maybe some others I'm forgetting too).

**Review by gardner48 — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/dealii/package.py (hunk 2 above)
Comment (balos1) [reply var/spack/repos/builtin/packages/dealii/package.py:209]: IIRC they already pin the version in their spack packages, but we will see what the CI reports back.

**Review by gardner48 — APPROVED** [review]

**Comment by balos1** [comment]
@alalazo CI is all passing now.

**Review by tldahlgren — APPROVED** [review]

**Comment by tldahlgren** [comment]
@alalazo Looks like your change request is blocking the merge.

**Review by alalazo — APPROVED** [review]

**Pull request merged**